extensions-contrib/opentsdb-emitter/src/main/java/io/druid/emitter/opentsdb/OpentsdbSender.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.emitter.opentsdb;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.WebResource;
+import io.druid.java.util.common.logger.Logger;
+
+import javax.ws.rs.core.MediaType;
+import java.util.ArrayList;
+import java.util.List;
+
+public class OpentsdbSender
+{
+  /**
+   * @see <a href="http://opentsdb.net/docs/build/html/api_http/put.html">Opentsdb - /api/put</a>
+   */
+  private static final String PATH = "/api/put";
+  private static final Logger log = new Logger(OpentsdbSender.class);
+
+  private final int batchSize;
+  private final List<OpentsdbEvent> events;
+  private final Client client;
+  private final WebResource webResource;
+
+  public OpentsdbSender(String host, int port, int connectionTimeout, int readTimeout, int batchSize)
+  {
+    this.batchSize = batchSize;
+    events = new ArrayList<>(batchSize);
+
+    client = Client.create();
+    client.setConnectTimeout(connectionTimeout);
+    client.setReadTimeout(readTimeout);
+    webResource = client.resource("http://" + host + ":" + port + PATH);
```

Comment:
How difficult would it be to add TLS support for this emitter? (Does TSDB support that to begin with?)

Reply:
As of yet, OpenTSDB lacks an authentication and access control system. And there is no HTTPS API support in OpenTSDB.

Reply:
Cool, thanks for checking on that